### PR TITLE
Add opensearch deployment configuration

### PR DESCRIPTION
**Pull request by AlexAvlonitis**
Please feel free to add any changes/refactorings straight to this PR

Resources: 
- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearch_domain
- https://medium.com/heyjobs-tech/how-to-create-an-opensearch-cluster-using-terraform-926b4a62b489

Note:

We need to choose between auth methods: IAM ARN, or master user/pass . This PR uses the master user/pass method

trello card: https://trello.com/c/MgjzeKGX/1326-hook-up-opensearch-in-aws

### CODE REVIEW

File: terraform/deployments/opensearch/security_groups.tf
Comment (rtrinque) [inline terraform/deployments/opensearch/security_groups.tf:1]: Should we limit the source that access the opensearch db to the eks cluster using data.tfe_outputs.cluster_infrastructure.nonsensitive_values.node_security_group_id as in https://github.com/alphagov/govuk-infrastructure/blob/main/terraform/deployments/govuk-publishing-infrastructure/security.tf ?

Comment (AlexAvlonitis) [reply terraform/deployments/opensearch/security_groups.tf:1]: Yup agreed, changed it, a question about this line here -> https://github.com/alphagov/govuk-infrastructure/pull/1288/files#diff-252631d8e214e762d830b65fa28508cf4ae7282f8c559c9c408cdb3a06a70660R7

`data.terraform_remote_state.infra_security_groups.outputs.sg_opensearch_id` I set up this sg_opensearch_id, but do we need to create it somehow beforehand?

Comment (AlexAvlonitis) [reply terraform/deployments/opensearch/security_groups.tf:1]: Good point, change it.

Added this line here -> https://github.com/alphagov/govuk-infrastructure/pull/1288/files#diff-252631d8e214e762d830b65fa28508cf4ae7282f8c559c9c408cdb3a06a70660R7

Do we need to create that security group beforehand from somewhere? the `sg_opensearch_id`

Comment (rtrinque) [reply terraform/deployments/opensearch/security_groups.tf:1]: yep, I've added definition for the security group

Comment (AlexAvlonitis) [reply terraform/deployments/opensearch/security_groups.tf:1]: Great stuff, I noticed that we probably also need to define this  `data.terraform_remote_state.infra_vpc.outputs.vpc_id` in the data.tf file, as the CI is complaining.

File: terraform/deployments/opensearch/main.tf [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+}
+
+locals {
+  domain            = "${var.service}-engine"
+  custom_domain     = "${local.domain}.${data.aws_route53_zone.opensearch.name}"
+  subnet_ids        = data.terraform_remote_state.infra_networking.outputs.private_subnet_rds_ids
```
Comment (AlexAvlonitis) [inline terraform/deployments/opensearch/main.tf:20]: This one might need some 👀  too. I have used the rds subnet ids: example here: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/opensearch_domain#vpc-based-opensearch

**Review by rtrinque — APPROVED** [review]
Looks good

**Pull request merged**